staging/src/k8s.io/apimachinery/pkg/util/json/json.go

```diff
@@ -119,3 +120,20 @@ func convertNumber(n json.Number) (interface{}, error) {
 	// An overflow will return an error
 	return n.Float64()
 }
+
+// Sniff does a constant-time inspection of the provided bytes and returns whether or not they may
+// contain JSON. If unknown is true, the determination could be wrong.
+func Sniff(src []byte) (json bool, unknown bool) {
+	if len(src) > 32 {
+		src = src[:32]
+	}
```

Comment:
does the 32 byte head limit correlate to some other limit? what would the impact of making this more lenient be (1k or something)?

Reply:
It's arbitrary, only setting of cap on how much of the input might be scanned before it makes a decision. It could be higher, I imagine the overwhelmingly common case is zero or nearly zero bytes of preceding whitespace.